src/graph/network.cc

```diff
@@ -807,6 +815,9 @@ DGL_REGISTER_GLOBAL("network._CAPI_FastPull")
     // Copy local data
 #pragma omp parallel for
     for (int64_t i = 0; i < local_ids.size(); ++i) {
+      CHECK_GE(ID_size*row_size, local_ids_orginal[i] * row_size + row_size);
+      CHECK_GE(data_size, local_ids[i] * row_size + row_size);
+      CHECK_GE(local_ids[i], 0);
```

Comment:
Does this checking will hurt our performance?

Reply:
most likely not. if node data is large, it doesn't matter much. If the node data is small (e.g., just a float), it might hurt performance. However, it's not efficient to use memcpy to copy a scalar either.